### PR TITLE
en.nepnep: hot updates page

### PR DESCRIPTION
**Pull request by yongsu-l**
copied a lot of the code from what was already here such as using a static cache and refactored things along the way.

I couldn't find references to `.SearchResult > .SearchResultCover` in their website so I'd assumed it was removed

Checklist:
- [x] Updated source's version for individual source changes
- [x] Updated all sources' versions for template changes
- [x] Set appropriate `nsfw` value
- [x] Did not change `id` even if a source's name or language were changed
- [x] Tested the modifications by running it on the simulator or a test device

### CODE REVIEW

**Comment by yongsu-l** [comment]
rebased and fixed!

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -298,3 +298,9 @@ Checklist:
 - [ ] Set appropriate `nsfw` value
 - [ ] Did not change `id` even if a source's name or language were changed
 - [ ] Tested the modifications by running it on the simulator or a test device 
+
+## Common Errors
+
+### `wasm32-unknown-unknown` target may not be installed
+
+Run `rustup target add wasm32-unknown-unknown --toolchain nightly`. Ref: https://github.com/rustwasm/wasm-bindgen/issues/929#issuecomment-427124646
```
Comment (Skittyblock) [inline .github/CONTRIBUTING.md:306]: this should be added in a separate pr rather than in a source update pr

it might be better to add some more info at the start about how to properly set up the dev environment

**Review by Skittyblock — CHANGES_REQUESTED** [review]
looks like you're using spaces, make sure to run `cargo fmt` to format everything properly, and run `cargo +nightly clippy` to see if there are any other warnings you should fix

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (yongsu-l) [reply .github/CONTRIBUTING.md]: sgtm

File: src/rust/en.nepnep/src/parser.rs [hunk 2]
```diff
@@ -8,6 +8,36 @@ use super::helper::{chapter_image, chapter_url_encode};
...
+		cover,
+		title,
+		author: String::new(),
+		artist: String::new(),
+		description: String::new(),
+		url,
+		categories: Vec::new(),
+		status: MangaStatus::Unknown,
+		nsfw: MangaContentRating::Safe,
+		viewer: MangaViewer::Default,
```
[the remark this answers is not in the document]
Comment (yongsu-l) [reply src/rust/en.nepnep/src/parser.rs]: sick didn't know we could do this

**Comment by yongsu-l** [comment]
would it be difficult to make `cargo +nightly test` work? i ran into issues w/ conflicting `std`

File: src/rust/en.nepnep/src/lib.rs [hunk 3]
```diff
@@ -157,9 +148,46 @@ fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
...
+		"Hot Updates" => {
+			if unsafe { HOT_UPDATE_RID } < 0 {
+				initialize_hot_update()
+			}
+		}
+		_ => {
+			println!("Received unexpected listing: {}", listing_name);
+			return Err(aidoku::error::AidokuError {
+				reason: aidoku::error::AidokuErrorKind::Unimplemented,
+			});
```
Comment (Skittyblock) [inline src/rust/en.nepnep/src/lib.rs:170]: honestly probably better to do this instead but it's a pretty minor issue. AidokuError constructing is kinda ugly and it's not really a relevant error.
```suggestion
			panic!("Received unexpected listing: {}", listing_name);
```

**Comment by Skittyblock** [comment]
> would it be difficult to make `cargo +nightly test` work? i ran into issues w/ conflicting `std`

honestly yeah that's a good question. it'd be good if we had the ability to create tests for sources, but the std error is kinda annoying and I'm not sure how to fix it without looking into it more. I've been meaning to redo a lot of the source stuff, so I might look into making things better then.

**Review by Skittyblock — APPROVED** [review]
